File: statelibrary/src/main/java/com/github/nukc/stateview/UriUtils.java
package com.github.nukc.stateview;

import android.content.Context;
import android.net.Uri;

/**
 * Created by Administrator on 2017/09/19.
 */

public class UriUtils {
    private static Uri uri;
    private static String path = Constant.STRINGDEFAULT;
    private static int resId = Constant.INTDEFAULT;

    public static Uri getUri(int mode, String url, Context context) {
        path = url;
        return UriTool(mode, context);
    }

    public static Uri getUri(int mode, int localResource, Context context) {
        resId = localResource;
        return UriTool(mode, context);
    }

    public static Uri UriTool(int mode, Context context) {
        if (mode == Constant.ONLINEPIC && path != Constant.STRINGDEFAULT) {
            uri = Uri.parse(path);
        } else if (mode == Constant.LOCALRESOURCE && resId != Constant.INTDEFAULT) {
            uri = Uri.parse("res://" + context.getPackageName() + "/" + resId);
        }

        return uri;
    }

}

File: app/src/main/java/com/xiangyao/train/utils/UriUtils.kt
package com.xiangyao.train.utils

import android.net.Uri

import com.xiangyao.train.base.DemoApplication

/**
 * Created by xiangyao on 2017/9/24.
 */

object UriUtils {
    private var uri: Uri? = null
    private var path = Constant.STRINGDEFAULT
    private var resId = Constant.INTDEFAULT

    fun getUri(mode: Int, url: String): Uri? {
        path = url
        return UriTool(mode)
    }

    fun getUri(mode: Int, localResource: Int): Uri? {
        resId = localResource
        return UriTool(mode)
    }

    fun UriTool(mode: Int): Uri? {
        if (mode == Constant.ONLINEPIC && path !== Constant.STRINGDEFAULT) {
            uri = Uri.parse(path)
        } else if (mode == Constant.LOCALRESOURCE && resId != Constant.INTDEFAULT) {
            uri = Uri.parse("res://" + DemoApplication.appContext!!.packageName + "/" + resId)
        }

        return uri
    }


}
